src/components/modal/modal.tsx

```diff
@@ -134,6 +130,7 @@ export class Modal
     setUpLoadableComponent(this);
     // when modal initially renders, if active was set we need to open as watcher doesn't fire
     if (this.open) {
+      onToggleOpenCloseComponent(this);
```

Comment:
Calling this here will emit these events on load of the component. Is there some history on why we are emitting on load here? Usually for components, we don't emit events initially, we start emitting after load on user interaction.

If this is intentional, the test will need to be updated to expect the event to be called initially.

Reply:
The discussion on this is moved to #4398.